src/CRSMatrixView.hpp

```diff
@@ -343,6 +343,38 @@ class CRSMatrixView : protected SparsityPatternView< COL_TYPE, INDEX_TYPE >
     }
   }
 
+  /**
+   * @brief Add to the given entries, the entries must already exist in the matrix.
+   * @details This method uses a binary search to find the entries in the row to add to.
+   *   This makes the method O( nCols * log( numNonZeros( @p row ) ) ) and is therefore best
+   *   to use when @p nCols is much less than numNonZeros( @p row ).
+   * @tparam AtomicPolicy the policy to use when adding to the values.
+   * @param row The row to access.
+   * @param cols The columns to add to, must be sorted, unique and of length @p nCols.
+   * @param vals The values to add, of length @p nCols.
+   * @param nCols The number of columns to add to.
+   */
+  template< typename AtomicPolicy >
+  LVARRAY_HOST_DEVICE inline
+  void addToRowBinarySearchUnsorted( INDEX_TYPE const row,
```

Comment:
The columns don't need to be sorted in this method, should be useful.

Reply:
I think this is the one that is useful for our assembly routines.